doc/builders/images/dockertools.section.md

```diff
@@ -394,3 +394,114 @@ buildImage {
   };
 }
 ```
+
+## buildNixShellImage {#ssec-pkgs-dockerTools-buildNixShellImage}
+
+Create a Docker image that sets up the `nix-shell` environment of a derivation. With Docker, this environment is similarly sandboxed as if the derivation were built with `nix-build`, with a major difference being that access to the internet is allowed. It additionally also behaves like an interactive `nix-shell`, running things like `shellHook` and setting an interactive prompt. If the derivation is fully buildable (i.e. `nix-build` can be used on it), running `buildDerivation` inside such a Docker image will build the derivation, with all its outputs being available in the correct `/nix/store` paths, pointed to by the respective environment variables like `$out`, etc.
+
+::: {.warning}
+The behavior doesn't match nix-shell exactly and this function may not work correctly for e.g. fixed-output derivations, CA derivations, impure derivations and other special types of derivations.
```

Comment:
```suggestion
The behavior doesn't match nix-shell or nix-build exactly and this function is known not to work correctly for e.g. fixed-output derivations, CA derivations, impure derivations and other special types of derivations.
```

These have extra behavior that don't make much sense to implement, so we can confidently state that they're broken.